postproengine/reynoldsstress.py

```diff
@@ -123,11 +142,140 @@ def execute(self, verbose=False):
+            self.parent = parent
+            print('Initialized '+self.actionname+' inside '+parent.name)
+            return
+
+        def execute(self):
+            include_radial = self.actiondict['include_radial']
+            iplanes = self.actiondict['iplane']
+
+            if iplanes == None:
+                iplanes = list(range(0,self.parent.dbReAvg['x'].shape[0]))
+
+            turbulent_transport_list = ['u_avg_uu_avg','u_avg_uv_avg','u_avg_uw_avg',\
+                                        'v_avg_uv_avg','v_avg_vv_avg','v_avg_vw_avg',\
+                                        'w_avg_uw_avg','w_avg_vw_avg','w_avg_ww_avg']
+                                        
+            if include_radial:
+                self.parent.dbReAvg['ux_avg_uxur_avg'] = np.zeros_like(self.parent.dbReAvg['uv_avg'])
+
+            for term in turbulent_transport_list:
+                self.parent.dbReAvg[term] = np.zeros_like(self.parent.dbReAvg['uv_avg'])
+
+            for iplaneiter, iplane in enumerate(iplanes):
+                u_avg  = self.parent.dbReAvg['velocityx_avg'][iplane,:,:]
+                v_avg  = self.parent.dbReAvg['velocityy_avg'][iplane,:,:]
+                w_avg  = self.parent.dbReAvg['velocityz_avg'][iplane,:,:]
+
+                uu_avg  = self.parent.dbReAvg['uu_avg'][iplane,:,:]
+                uv_avg  = self.parent.dbReAvg['uv_avg'][iplane,:,:]
+                uw_avg  = self.parent.dbReAvg['uw_avg'][iplane,:,:]
+
+                vv_avg  = self.parent.dbReAvg['vv_avg'][iplane,:,:]
+                vw_avg  = self.parent.dbReAvg['vw_avg'][iplane,:,:]
+                ww_avg  = self.parent.dbReAvg['ww_avg'][iplane,:,:]
+
+
+                self.parent.dbReAvg['u_avg_uu_avg'][iplane,:,:] = u_avg * self.parent.dbReAvg['uu_avg'][iplane,:,:]
+                self.parent.dbReAvg['u_avg_uv_avg'][iplane,:,:] = u_avg * self.parent.dbReAvg['uv_avg'][iplane,:,:]
+                self.parent.dbReAvg['u_avg_uw_avg'][iplane,:,:] = u_avg * self.parent.dbReAvg['uw_avg'][iplane,:,:]
+
+                self.parent.dbReAvg['v_avg_uv_avg'][iplane,:,:] = v_avg * self.parent.dbReAvg['uv_avg'][iplane,:,:]
+                self.parent.dbReAvg['v_avg_vv_avg'][iplane,:,:] = v_avg * self.parent.dbReAvg['vv_avg'][iplane,:,:]
+                self.parent.dbReAvg['v_avg_vw_avg'][iplane,:,:] = v_avg * self.parent.dbReAvg['vw_avg'][iplane,:,:]
+
+                self.parent.dbReAvg['w_avg_uw_avg'][iplane,:,:] = w_avg * self.parent.dbReAvg['uw_avg'][iplane,:,:]
+                self.parent.dbReAvg['w_avg_vw_avg'][iplane,:,:] = w_avg * self.parent.dbReAvg['vw_avg'][iplane,:,:]
+                self.parent.dbReAvg['w_avg_ww_avg'][iplane,:,:] = w_avg * self.parent.dbReAvg['ww_avg'][iplane,:,:]
+
+                if include_radial:
+                    self.parent.dbReAvg['ux_avg_uxur_avg'][iplane,:,:] = u_avg * self.parent.dbReAvg['uxur_avg'][iplane,:,:]
+
+            # Overwrite picklefile
+            if len(self.parent.pklfile)>0:
```

Comment:
Do we want to save all these to the same pickle file or is there a concern about the file getting too large? Should we separate turbulent stresses and fluxes?